lib/ansible/module_utils/tacp_ansible/tacp_utils.py

```diff
@@ -129,11 +129,12 @@ def get_filters_query_string(self, **kws):
 
                 value = value[0]
                 if op in ('=in=', '=out='):
-                    value = ','.join(map(str, value))
+                    value = ','.join(map(lambda v: '"{}"'.format(v), value))
                     value = '({})'.format(value)
             else:
                 op = '=='
-                value = v
+                value = '"{}"'.format(v)
```

Comment:
Python wise looks good. I dont know whether it fixes the issue or not, depends on the api. Have you at least tested the scenarios in an interpreter?

Reply:
yes, i ran all three modules with this fix and haven't run into the bugs anymore.